tasks/argocd/src/main/java/com/walmartlabs/concord/plugins/argocd/ArgoCdTask.java

```diff
@@ -56,51 +57,73 @@ public ArgoCdTask(Context context, LockService lockService, ApiClient apiClient)
-                return processCreateApplicationSetAction((TaskParams.CreateUpdateApplicationSetParams) params);
-            }
-            default: {
-                throw new IllegalArgumentException("Unsupported action type: " + params.action());
+        try {
+            TaskResult taskResult;
+            record(params, EventStatus.START);
+            switch (params.action()) {
+                case GET: {
+                    taskResult = processGetAction((TaskParams.GetParams) params);
+                    break;
+                }
+                case SYNC: {
+                    taskResult = processSyncAction((TaskParams.SyncParams) params);
+                    break;
+                }
+                case DELETE: {
+                    taskResult = processDeleteAction((TaskParams.DeleteAppParams) params);
+                    break;
+                }
+                case PATCH: {
+                    taskResult = processPatchAction((TaskParams.PatchParams) params);
+                    break;
+                }
+                case UPDATESPEC: {
+                    taskResult = processUpdateSpecAction((TaskParams.UpdateSpecParams) params);
+                    break;
+                }
+                case SETPARAMS: {
+                    taskResult = processSetParamsAction((TaskParams.SetAppParams) params);
+                    break;
+                }
+                case CREATE: {
+                    taskResult = processCreateAction((TaskParams.CreateUpdateParams) params);
+                    break;
+                }
+                case GETPROJECT: {
+                    taskResult = processGetProjectAction((TaskParams.GetProjectParams) params);
+                    break;
+                }
+                case CREATEPROJECT: {
+                    taskResult = processProjectCreateAction((TaskParams.CreateProjectParams) params);
+                    break;
+                }
+                case DELETEPROJECT: {
+                    taskResult = processProjectDeleteAction((TaskParams.DeleteProjectParams) params);
+                    break;
+                }
+                case GETAPPLICATIONSET: {
+                    taskResult = processGetApplicationSetAction((TaskParams.GetApplicationSetParams) params);
+                    break;
+                }
+                case DELETEAPPLICATIONSET: {
+                    taskResult = processDeleteApplicationSetAction((TaskParams.DeleteApplicationSetParams) params);
+                    break;
+                }
+                case CREATEAPPLICATIONSET: {
+                    taskResult = processCreateApplicationSetAction((TaskParams.CreateUpdateApplicationSetParams) params);
+                    break;
+                }
+                default: {
+                    throw new IllegalArgumentException("Unsupported action type: " + params.action());
+                }
             }
+            record(params, EventStatus.COMPLETE);
```

Comment:
there are no results put in for completion event?

Reply:
Updated